fix: ELSEIF handling

bendavies opened the pull request:
primarily this resolves the "WARNING: unclosed parentheses or section" error that occurred when parsing SQL statements containing ELSEIF keywords in IF/ELSEIF/ELSE/END IF blocks.

second, I noticed and fixed that AND/OR weren't handled correctly inside IF blocks, where new lines were added incorrectly.

bendavies committed:
fix: ELSEIF handling in SQL formatter to resolve unclosed parentheses warning

This resolves the "WARNING: unclosed parentheses or section" error that occurred when parsing SQL statements containing ELSEIF keywords in IF/ELSEIF/ELSE/END IF blocks.

bendavies committed:
fix: AND/OR newline behavior within IF conditions

- Add context tracking to detect when inside IF conditions
- Skip newlines for AND/OR tokens within IF conditions while preserving
  newline behavior for AND/OR in WHERE clauses and other contexts
- This improves readability of formatted SQL with complex IF statements

mvorisek commented on src/Tokenizer.php:
```diff
@@ -104,6 +104,7 @@ final class Tokenizer
         'DUPLICATE',
         'DYNAMIC',
         'ELSE',
+        'ELSEIF',
```
This is a little problematic as it would incorrectly highlight "elseif" as reserved but it is not reserved in every standard SQL. Example: `Select 'foo' as ELSEIF`.

Also, which DB use `ELSEIF`?

And also note PostgreSQL use `ELSIF`.

mvorisek replied:
I think at least something like outlined in https://github.com/doctrine/sql-formatter/issues/118 is needed. We need to parse some "IF ELSE/ELSIF/ELSEIF ELSE END IF" AST tree in order to be able to parse this correctly without knowing the exact DB type/grammar.

bendavies replied:
> This is a little problematic as it would incorrectly highlight "elseif" as reserved but it is not reserved in every standard SQL. Example: `Select 'foo' as ELSEIF`.
> 
> Also, which DB use `ELSEIF`?
> 
> And also note PostgreSQL use `ELSIF`.

that's a very astute observation. 

We have functions in postgres 16 using `ELSEIF`. I wasn't aware it was non standard.
https://www.postgresql.org/docs/current/plpgsql-control-structures.html
> The key word ELSIF can also be spelled ELSEIF.

I'm happy to migrate this PR to use ELSIF.

mvorisek replied:
> We have functions in postgres 16 using `ELSEIF`. I wasn't aware it was non standard. https://www.postgresql.org/docs/current/plpgsql-control-structures.html
> 
> > The key word ELSIF can also be spelled ELSEIF.
> 
> I'm happy to migrate this PR to use ELSIF.

Me neither, as you use 2 ELSEIF in test case already, simply modify 1 to ELSIF. We should support both.

bendavies replied:
done

bendavies committed:
feat: add support for ELSIF keyword alongside ELSEIF

mvorisek commented on src/SqlFormatter.php:
```diff
@@ -332,9 +347,12 @@ public function format(string $string, string $indentString = '  '): string
                 }
             } elseif ($token->isOfType(Token::TOKEN_TYPE_RESERVED_NEWLINE)) {
-                // Newline reserved words start a new line
+                // Newline reserved words start a new line, except for AND/OR within IF conditions
```
There are more contexts where boolean expressions can be used the same way, at least CASE WHEN. Not sure if this optimization is welcomed in this PR.

bendavies replied:
done

The behavior is now:
 - IF/ELSEIF/ELSIF conditions: AND/OR stay on the same line
 - CASE WHEN conditions: AND/OR stay on the same line
 - WHERE/HAVING clauses: AND/OR still get newlines (unchanged - this is probably desired for readability)
 - JOIN ON conditions: AND/OR still get newlines (unchanged)

bendavies committed:
feat: extend AND/OR formatting optimization to CASE WHEN conditions

- Apply same-line AND/OR formatting to CASE WHEN conditions
- Maintain newline behavior for WHERE/HAVING/JOIN clauses
